js/DataService.js

```diff
@@ -34,140 +67,63 @@ angular.module('PathOfDamage')
-            }
+            reduction: new Table("Additional Physical Damage Reduction", function (totals, value) {
+              totals.total = Math.min(totals.total + value.value, 90);
+            })
           }
         },
         taken: {
           name: "Damage Taken",
           description: "After damage mitigation, modifiers to damage taken are applied. Flat amounts (±X Damage taken from Y, like Astramentis) are applied first, then the sum of all increases/reductions (% increased/reduced X Damage taken, like Fortify, Shock, and Abyssus) and lastly more/less multipliers (% more/less X Damage taken, like Arctic Armour).",
           tables: {
-            flat: {
-              name: "Flat Increase",
-              calcTotal: function () {
-                var subTotals = {};
-                this.values.forEach(function (flat) {
-                  if (flat.enabled && flat.value && flat.types) {
-                    flat.types.forEach(function (type) {
-                      subTotals[type] = subTotals[type] + flat.value || flat.value;
-                    });
-                  }
-                });
-                this.subTotals = subTotals;
-              },
-              getTotal: function () {
-                return this.subTotals.physical || 0;
-              }
-            },
-            increased: {
-              name: "Increased Damage Taken",
-              calcTotal: function () {
-                var subTotals = {};
-                this.values.forEach(function (increased) {
-                  if (increased.enabled && increased.value && increased.types) {
-                    increased.types.forEach(function (type) {
-                      subTotals[type] = Math.max(subTotals[type] + increased.value || increased.value, -100);
-                    });
-                  }
-                });
-                this.subTotals = subTotals;
-              },
-              getTotal: function () {
-                return this.subTotals.physical || 0;
-              }
-            },
-            more: {
-              name: "More Damage Taken",
-              calcTotal: function () {
-                var subTotals = {};
-                this.values.forEach(function (more) {
-                  if (more.enabled && more.value && more.types) {
-                    more.types.forEach(function (type) {
-                      var multiplier = 1 + more.value / 100;
-                      var newTotal = (subTotals[type] + 100) * multiplier - 100;
-                      subTotals[type] = Math.max(newTotal || more.value, -100);
-                    });
-                  }
-                });
-                this.subTotals = subTotals;
-              },
-              getTotal: function () {
-                return this.subTotals.physical || 0;
-              }
-            }
+            flat: new TypedTable("Flat Increase", function (totals, value, type) {
+              totals[type] = totals[type] + value || value;
```

Comment:
If we initialize totals such that totals[type] is null for all the types, this could be simplified to just totals[type] + value, since number + null = number